src/unit-workspace/unit.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@nestjs/common';
+import { currentDateTime } from '@us-epa-camd/easey-common/utilities/functions';
+
+import { UnitWorkspaceRepository } from './unit.repository';
+import { UnitBaseDTO, UnitDTO } from '../dtos/unit.dto';
+import { EntityManager } from 'typeorm';
+
+@Injectable()
+export class UnitWorkspaceService {
+  constructor(
+    private readonly repository: UnitWorkspaceRepository,
+    private readonly entityManager: EntityManager,
+  ) {}
+
+  async getUnit(locId: string, unitId: number,): Promise<UnitDTO> {
+    const unitDetails = await this.getUnitDetails(unitId);
+    return unitDetails && unitDetails.length > 0 ? unitDetails[0] : null;
+  }
+
+  async getUnits(locId: string, unitId: number,): Promise<UnitDTO[]> {
+    return await this.getUnitDetails(unitId);
+  }
+
+  async updateUnit(
+    locationId: string,
+    unitId: number,
+    payload: UnitBaseDTO,
+    userId: string,
+    isImport = false,
+  ): Promise<UnitDTO> {
+    const unit = await this.repository.findOneBy({ id: unitId });
+
+    unit.nonLoadBasedIndicator = payload.nonLoadBasedIndicator;
+    unit.userId = userId;
+    unit.updateDate = currentDateTime();
+
+    await this.repository.save(unit);
+
+    const unitDetails = await this.getUnitDetails(unitId);
+    return unitDetails && unitDetails.length > 0 ? unitDetails[0] : null;
+  }
+
+  private async getUnitDetails(unitId: number): Promise<UnitDTO[]> {
+    const sql = `
+        SELECT
+            unt.UNIT_ID as "id",
+            camdUnt.UNITID as "unitid",  -- From camd.UNIT
+            unt.NON_LOAD_BASED_IND as "nonLoadBasedIndicator",  -- From camdecmpswks.UNIT
+            camdUnt.SOURCE_CATEGORY_CD as "sourceCategoryCd",  -- From camd.UNIT
+            to_char(camdUnt.COMM_OP_DATE, 'yyyy-mm-dd') AS "commOpDate",  -- From camd.UNIT
+            to_char(camdUnt.COMR_OP_DATE, 'yyyy-mm-dd') AS "comrOpDate",  -- From camd.UNIT
+            (
+                SELECT ubt.UNIT_TYPE_CD
```

Comment:
I don't think that it is possible but you may want to put a MAX() around ubt.UNIT_TYPE_CD just in case non UNIT_BOILER_TYPE rows exists for a particular unit.